crates/swc_ecma_transforms_compat/tests/es2017_async_to_generator.rs

```diff
@@ -1370,10 +1370,10 @@ let TestClass = {
     name: 'John Doe',
     testMethodFailure () {
         return new Promise(function() {
-            var _ref = _asyncToGenerator((function*(resolve) {
+            var _ref = _asyncToGenerator(function*(resolve) {
```

Comment:
we need to pass this throught `FnEnvHoister` or something like [babel](https://babeljs.io/repl#?browsers=chrome%2050&build=&builtIns=false&corejs=3.6&spec=false&loose=true&code_lz=DYUwLgBAKiDOYGFgENawgXggbwFAQgDtkBbEALggCIApAewAtCIAROkKgGnwjDjACy4BnQAmAMWQBLYAFcATiAAUAShw8CisAuaEQAdwgAFeXRJTYy1AE9CAYwhLFsOsABuINRgB86gv4g7OkIXUAA6YDoAcyUwBgsVAG4Nf0swKCkyOlkwJzhXD04IAEYABnKklIBfSoIq3CrEoA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact&prettier=false&targets=&version=7.16.9&externalPlugins=&assumptions=%7B%22ignoreFunctionLength%22%3Atrue%7D)